kedro_vertexai/config.py

```diff
@@ -127,18 +128,33 @@ class RunConfig(BaseModel):
     resources: Optional[Dict[str, ResourcesConfig]] = dict(
         __default__=ResourcesConfig(cpu="500m", memory="1024Mi")
     )
+    node_selectors: Optional[Dict[str, Dict[str, str]]] = {}
     dynamic_config_providers: Optional[List[DynamicConfigProviderConfig]] = []
     mlflow: Optional[MLFlowVertexAIConfig] = None
 
-    def resources_for(self, node):
-        if node in self.resources.keys():
-            result = self.resources["__default__"].dict()
-            result.update(
-                {k: v for k, v in self.resources[node].dict().items() if v is not None}
-            )
-            return result
-        else:
-            return self.resources["__default__"].dict()
+    def resources_for(self, node: str, tags: Optional[set] = set()):
+        default_config = self.resources["__default__"].dict()
+        return self._config_for(node, tags, self.resources, default_config)
+
+    def node_selectors_for(self, node: str, tags: Optional[set] = set()):
+        return self._config_for(node, tags, self.node_selectors)
+
+    @staticmethod
+    def _config_for(
+        node: str, tags: set, params: dict, default_config: Optional[dict] = None
+    ):
+        names = [*tags, node]
```

Comment:
From the usage here I see that tags are not really optional, won't work with set to None (see node_selectors_for and resources_for). Setting collections as default parameters is a bad practice - if anything changes that collection, the change will be present in all of the subsequent calls as default.

Reply:
Default parameters changed to `None` ,  added `tags = tags or set()`.